compiler/noirc_frontend/src/elaborator/types.rs

```diff
@@ -92,7 +108,7 @@ impl<'context> Elaborator<'context> {
             Unspecified => Type::Error,
             Error => Type::Error,
             Named(path, args, _) => self.resolve_named_type(path, args),
-            TraitAsType(path, args) => self.resolve_trait_as_type(path, args, kind),
+            TraitAsType(path, args) => self.resolve_trait_as_type(path, args),
```

Comment:
Won't any global we find be evaluated to a `u32` constant, thus not having a kind? And then the kind is check after this match statement with the `NamedGeneric` check?

Reply:
A global would resolve to a `Type::Constant` holding it's value, so we'll need to ensure a numeric type is expected in that case. The NamedGeneric check wouldn't catch this case since it is not a NamedGeneric. We'll need to check every type, not just generics.

Reply:
Discussed on slack, this check will be generalized in a follow-up when we remove implicit numeric generics https://github.com/noir-lang/noir/issues/5156